webapp/lib/controller.dart

```diff
@@ -108,71 +119,117 @@ void command(UIAction action, Data actionData) {
+  view.contentView.needsReplyAndEscalateTimeseries.updateChart([data]);
+
+  data = new Map.fromIterable(selectedProjectNeedsReplyDataList,
+    key: (item) => (item as model.NeedsReplyData).datetime,
+    value: (item) => (item as model.NeedsReplyData).needsReplyMoreThan24h);
+  view.contentView.needsReplyMoreThan24hTimeseries.updateChart([data]);
+
+  data = new Map.fromIterable(selectedProjectNeedsReplyDataList,
+    key: (item) => (item as model.NeedsReplyData).datetime,
+    value: (item) => (item as model.NeedsReplyData).needsReplyAndEscalateMoreThan24hCount);
+  view.contentView.needsReplyAndEscalateMoreThan24hTimeseries.updateChart([data]);
+
+  DateTime latestDateTime = data.keys.reduce((dt1, dt2) => dt1.isAfter(dt2) ? dt1 : dt2);
+  var latestData = selectedProjectNeedsReplyDataList.firstWhere((d) => d.datetime == latestDateTime, orElse: () => null);
+
+  view.contentView.needsReplyLatestValue.updateChart('${latestData.needsReplyCount}');
+  view.contentView.needsReplyAndEscalateLatestValue.updateChart('${latestData.needsReplyAndEscalateCount}');
+  view.contentView.needsReplyMoreThan24hLatestValue.updateChart('${latestData.needsReplyMoreThan24h}');
+  view.contentView.needsReplyAndEscalateMoreThan24hLatestValue.updateChart('${latestData.needsReplyAndEscalateMoreThan24hCount}');
+
+  view.contentView.needsReplyAgeHistogram.updateChart(latestData.needsReplyMessagesByDate);
+}
+
+void updateSystemEventsCharts(List<model.SystemEventsData> filteredSystemEventsDataList) {
+  var rapidProEventData = filteredSystemEventsDataList.where((eventData) =>
+      eventData.systemName == 'rapidpro_adapter' &&
+      eventData.project == view.contentView.projectSelectorView.selectedProject);
+  var pubsubEventData = filteredSystemEventsDataList.where((eventData) =>
+      eventData.systemName == 'pubsub_handler' &&
+      eventData.project == view.contentView.projectSelectorView.selectedProject);
+  
+  Map<DateTime, int> data = new Map.fromIterable(rapidProEventData,
+      key: (item) => (item as model.SystemEventsData).timestamp,
+      value: (item) => 1);
+  view.contentView.rapidProSystemEventTimeseries.updateChart([data]);
+
+  data = new Map.fromIterable(pubsubEventData,
+      key: (item) => (item as model.SystemEventsData).timestamp,
+      value: (item) => 1);
+  view.contentView.pubsubSystemEventTimeseries.updateChart([data]);
+}
+
+DateTime getFilteredDate(ChartFilterdata filterData) {
+  DateTime now = new DateTime.now();
+  DateTime filterDate;
+
+  switch (filterData.periodFilter) {
+  case ChartPeriodFilters.days1:
+    filterDate = now.subtract(Duration(days: 1));
+    break;
+  case ChartPeriodFilters.days8:
+    filterDate = now.subtract(Duration(days: 8));
+    break;
+  case ChartPeriodFilters.days15:
+    filterDate = now.subtract(Duration(days: 15));
+    break;
+  case ChartPeriodFilters.month1:
+    filterDate = now.subtract(Duration(days: 31));
+    break;
+  }
+
+  return filterDate;
+}
```

Comment:
nit: newline needed at end of file, Github's complaining
[you can set your editor to add this automatically so you don't need to always worry about it]

Reply:
Added newline. I'll setup my editor to this for me